core/state/statedb.go

```diff
@@ -1423,11 +1423,8 @@ func (s *StateDB) Commit(failPostCommitFunc func(), postCommitFuncs ...func() er
 			finishCh := make(chan struct{})
 
 			threads := gopool.Threads(len(s.stateObjectsDirty))
-			wg := sync.WaitGroup{}
```

Comment:
why it can be removed?

Reply:
there's a `read loop`, range for `tasksNums` times, makes it just work as synchronized, no need to `Wait` before the last return.